audit/format.go

```diff
@@ -49,28 +49,29 @@ func (f *AuditFormatter) FormatRequest(
 		return errwrap.Wrapf("error fetching salt: {{err}}", err)
 	}
 
+	var auth *logical.Auth
```

Comment:
Why are you instantiating these two variables? As best as I can tell these are actually masking an error down below where it's using auth/req instead of in.Auth/in.Request.

Reply:
The `req` variable is assigned to the value of `cp.(*logical.Request)`, which is then used for the `Hash()` to change the request to one with the values hashed, which is then in turn used as values for `AuditRequestEntry ` for `FormatRequest`. We didn't have to declare these before because `req` was part of the method's parameter and we were simply re-assigning that. We can do the same by re-assigning in.Request, but I decided to use new variables for the sake or readability. Same applies to `auth`.

Reply:
Per discussion -- assign these values to the input values at the top. That way if config.Raw is true, and they're not assigned in that block, we won't panic (otherwise we'll instantly panic for any raw audit backend).